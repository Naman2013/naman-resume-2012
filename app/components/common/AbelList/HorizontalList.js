import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './HorizontalList.style';

const HorizontalList = ({ list, theme, iconList }) => (
  <Fragment>
    <ul style={theme}>
      {list.map((entry, i) => (
        <li key={`horizontal-list-${entry}`}>
          {' '}
          {iconList[i] && <img className="bullet-image" src={iconList[i]} />}
          {entry}
        </li>
      ))}
    </ul>
    <style jsx>{style}</style>
  </Fragment>
);

HorizontalList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string).isRequired,
  theme: PropTypes.shape({}),
  iconList: PropTypes.arrayOf(PropTypes.string),
};

HorizontalList.defaultProps = {
  theme: {},
  iconList: [],
};

export default HorizontalList;
