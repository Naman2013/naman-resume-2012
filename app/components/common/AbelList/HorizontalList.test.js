import React from 'react';
import { shallow } from 'enzyme';
import HorizontalList from './HorizontalList';

describe('HorizontalList', () => {
  const list = ['foo', 'bar'];
  const shallowWrapper = shallow(<HorizontalList list={list} />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
