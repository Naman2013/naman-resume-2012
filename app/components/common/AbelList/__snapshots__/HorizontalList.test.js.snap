// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HorizontalList
    list={
      Array [
        "foo",
        "bar",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ul
          className="jsx-3148834294"
        >
          <li
            className="jsx-3148834294"
          >
            foo
          </li>
          <li
            className="jsx-3148834294"
          >
            bar
          </li>
        </ul>,
        <JSXStyle
          css={
            Array [
              "ul.jsx-3148834294{margin:0;padding:0;list-style-type:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;font-family:brandon-grotesque, sans-serif;color:#415671;text-transform:uppercase;border-top:1px solid #eaf0f9;border-bottom:1px solid #eaf0f9;box-shadow:inset 0px 5px 20px -5px #e0e0e0;}",
              "li.jsx-3148834294{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;font-weight:bold;padding:29.5px 0;padding-left:30px;font-size:11px;border-right:1px solid #eaf0f9;-webkit-letter-spacing:0.4px;-moz-letter-spacing:0.4px;-ms-letter-spacing:0.4px;letter-spacing:0.4px;}",
              "li.jsx-3148834294:last-child{border-right:none;}",
            ]
          }
          styleId="3148834294"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="jsx-3148834294"
            >
              foo
            </li>,
            <li
              className="jsx-3148834294"
            >
              bar
            </li>,
          ],
          "className": "jsx-3148834294",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "horizontal-list-foo",
            "nodeType": "host",
            "props": Object {
              "children": "foo",
              "className": "jsx-3148834294",
            },
            "ref": null,
            "rendered": "foo",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "horizontal-list-bar",
            "nodeType": "host",
            "props": Object {
              "children": "bar",
              "className": "jsx-3148834294",
            },
            "ref": null,
            "rendered": "bar",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "ul.jsx-3148834294{margin:0;padding:0;list-style-type:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;font-family:brandon-grotesque, sans-serif;color:#415671;text-transform:uppercase;border-top:1px solid #eaf0f9;border-bottom:1px solid #eaf0f9;box-shadow:inset 0px 5px 20px -5px #e0e0e0;}",
            "li.jsx-3148834294{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;font-weight:bold;padding:29.5px 0;padding-left:30px;font-size:11px;border-right:1px solid #eaf0f9;-webkit-letter-spacing:0.4px;-moz-letter-spacing:0.4px;-ms-letter-spacing:0.4px;letter-spacing:0.4px;}",
            "li.jsx-3148834294:last-child{border-right:none;}",
          ],
          "styleId": "3148834294",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ul
            className="jsx-3148834294"
          >
            <li
              className="jsx-3148834294"
            >
              foo
            </li>
            <li
              className="jsx-3148834294"
            >
              bar
            </li>
          </ul>,
          <JSXStyle
            css={
              Array [
                "ul.jsx-3148834294{margin:0;padding:0;list-style-type:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;font-family:brandon-grotesque, sans-serif;color:#415671;text-transform:uppercase;border-top:1px solid #eaf0f9;border-bottom:1px solid #eaf0f9;box-shadow:inset 0px 5px 20px -5px #e0e0e0;}",
                "li.jsx-3148834294{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;font-weight:bold;padding:29.5px 0;padding-left:30px;font-size:11px;border-right:1px solid #eaf0f9;-webkit-letter-spacing:0.4px;-moz-letter-spacing:0.4px;-ms-letter-spacing:0.4px;letter-spacing:0.4px;}",
                "li.jsx-3148834294:last-child{border-right:none;}",
              ]
            }
            styleId="3148834294"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="jsx-3148834294"
              >
                foo
              </li>,
              <li
                className="jsx-3148834294"
              >
                bar
              </li>,
            ],
            "className": "jsx-3148834294",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "horizontal-list-foo",
              "nodeType": "host",
              "props": Object {
                "children": "foo",
                "className": "jsx-3148834294",
              },
              "ref": null,
              "rendered": "foo",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "horizontal-list-bar",
              "nodeType": "host",
              "props": Object {
                "children": "bar",
                "className": "jsx-3148834294",
              },
              "ref": null,
              "rendered": "bar",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "ul.jsx-3148834294{margin:0;padding:0;list-style-type:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;font-family:brandon-grotesque, sans-serif;color:#415671;text-transform:uppercase;border-top:1px solid #eaf0f9;border-bottom:1px solid #eaf0f9;box-shadow:inset 0px 5px 20px -5px #e0e0e0;}",
              "li.jsx-3148834294{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;font-weight:bold;padding:29.5px 0;padding-left:30px;font-size:11px;border-right:1px solid #eaf0f9;-webkit-letter-spacing:0.4px;-moz-letter-spacing:0.4px;-ms-letter-spacing:0.4px;letter-spacing:0.4px;}",
              "li.jsx-3148834294:last-child{border-right:none;}",
            ],
            "styleId": "3148834294",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
